src/components/Api.js

```diff
@@ -0,0 +1,101 @@
+export default class Api {
+  constructor({ baseUrl, headers }, authToken, id, avatar, isLiked) {
+    this._baseUrl = baseUrl;
+    this._header = headers;
+    this._authToken = authToken;
+    this._id = id;
+    this._avatar = avatar;
+    this._isLiked = isLiked;
+  }
+
+  _checkResponse(res) {
+    // here is the code of the checking
+    console.log(888);
+    console.log(res);
+    return res.ok ? res.json() : Promise.reject(res);
+  }
+
+  _request(url, options) {
+    console.log(777);
+    return fetch(url, options);
+  }
```

Comment:
`_request` is not needed if it doesn't have `.then(this._checkResponse)` inside it


```js

  _request(url, options) {
    return fetch(url, options).then(this._checkResponse);
  }
```


And then you can remove all `.then(this._checkResponse);` from the code


```js


  getCardList() {
    return this._request(`${this._baseUrl}/cards`, {
      headers: this._header,
      method: "GET",
    });
  }

  getUserInfo() {
    return this._request(`${this._baseUrl}/users/me`, {
      headers: this._header,
      method: "GET",
    });
  }

```